### PR TITLE
Doc language edits - "Deep Learning" and "Guides"

### PR DESCRIPTION
**Pull request by yuxizama**
docs: language edits

### CODE REVIEW

File: docs/getting-started/concepts.md [hunk 1]
```diff
@@ -14,8 +14,8 @@ With Concrete ML, you can train a model on clear or encrypted data, then deploy
 
 1. **Quantization:** Quantization converts inputs, model weights, and all intermediate values of the inference computation to integer equivalents. More information is available [here](../explanations/quantization.md). Concrete ML performs this step in two ways depending on model type:
 
-   - During training (Quantization Aware Training)
-   - After training (Post-training Quantization)
+   - During training (Quantization Aware Training): Add explanation
+   - After training (Post Training Quantization): Add explanation
```
Comment (yuxizama) [inline docs/getting-started/concepts.md:18]: Can we add some simple explanations here about QAT and PTQ in the "concepts" doc?
In other docs whenever there's a mention of QAT and PTQ, I link here as a reference.

File: docs/deep-learning/fhe_assistant.md [hunk 2]
```diff
@@ -104,11 +104,11 @@ In the example above, the `x` and `y` layers need quantization before being conc
 z = torch.cat([self.quant_concat(x), self.quant_concat(y)])
 ```
 
-## Debugging compilation errors
+## Debugging compilation errors [TBD]
```
Comment (yuxizama) [inline docs/deep-learning/fhe_assistant.md:107]: This part seems redundant to the newly added "Common compilation errors", shall we delete or reframe it?

File: docs/deep-learning/fhe_assistant.md [hunk 3]
```diff
@@ -162,11 +162,11 @@ Function you are trying to compile cannot be compiled:
...
 
-On the other hand, `NoParametersFound` is encountered when using `rounding_threshold_bits`. When using this setting, the 16-bit accumulator limit is relaxed. However, reducing bit-width, or reducing the `rounding_threshold_bits`, or using  using the [`fhe.Exactness.APPROXIMATE`](../references/api/concrete.ml.torch.compile.md#function-compile_torch_model) rounding method can help.
+On the other hand, `NoParametersFound` occurs when using `rounding_threshold_bits`. With this setting, the 16-bit accumulator limit is relaxed. However, reducing bit-width, or reducing the `rounding_threshold_bits`, or using the [`fhe.Exactness.APPROXIMATE`](../references/api/concrete.ml.torch.compile.md#function-compile_torch_model) rounding method can help.
 
-### Fixing compilation errors
+### Fixing compilation errors [TBD]
```
Comment (yuxizama) [inline docs/deep-learning/fhe_assistant.md:169]: Same as above: This part seems redundant to the newly added "Common compilation errors", shall we delete or reframe it?

**Review by yuxizama — COMMENTED** [review]
Hey @zama-ai/machine-learning 
There are some changes needed in this PR, I marked them in comments. (Seems like I can not request changes as the author of the PR)
Please have a look. Thanks!

File: docs/deep-learning/onnx_support.md [hunk 4]
```diff
@@ -74,7 +76,7 @@ quantized_numpy_module = compile_onnx_model(
 
 ## Supported operators
 
-The following ONNX operators are supported for evaluation and conversion to an equivalent FHE circuit. Other operators were not implemented, either due to FHE constraints or because they are rarely used in PyTorch activations or scikit-learn models.
+Concrete ML support the following operators for evaluation and conversion to an equivalent FHE circuit. Other operators were not implemented either due to FHE constraints or because they are rarely used in PyTorch activations or scikit-learn models.
```
Comment (kcelia) [inline docs/deep-learning/onnx_support.md:79]: supports ?

File: docs/deep-learning/optimizing_inference.md [hunk 5]
```diff
@@ -1,21 +1,23 @@
...
 
 ## Circuit bit-width optimization
 
-[Quantization Aware Training](../explanations/quantization.md) and [pruning](../explanations/pruning.md) introduce specific hyper-parameters that influence the accumulator sizes. It is possible to chose quantization and pruning configurations that reduce the accumulator size. A trade-off between latency and accuracy can be obtained by varying these hyper-parameters as described in the [deep learning design guide](torch_support.md#configuring-quantization-parameters).
+[Quantization Aware Training](../explanations/quantization.md) and [pruning](../explanations/pruning.md) introduce specific hyper-parameters that influence the accumulator sizes. You can chose quantization and pruning configurations to reduce the accumulator size. To obtain a trade-off between latency and accuracy, you can manually set these hyper-parameters as described in the [deep learning design guide](torch_support.md#configuring-quantization-parameters).
```
Comment (kcelia) [inline docs/deep-learning/optimizing_inference.md:11]: Small remark, in the past, we agreed in the cml team to avoid as much as possible, sentence like you can, we do / will do... 
I don't if this rule changed

File: docs/deep-learning/optimizing_inference.md [hunk 6]
```diff
@@ -1,21 +1,23 @@
...
 
-Reducing the bit-width of the inputs to the Table Lookup (TLU) operations is a major source of improvements in the latency. Post-training, it is possible to leverage some properties of the fused activation and quantization functions expressed in the TLUs to further reduce the accumulator. This is achieved through the _rounded PBS_ feature as described in the [rounded activations and quantizers reference](../explanations/advanced_features.md#rounded-activations-and-quantizers). Adjusting the rounding amount, relative to the initial accumulator size, can bring large improvements in latency while maintaining accuracy.
+Reducing the bit-width of inputs to the Table Lookup (TLU) operations significantly improves latency. Post-training, you can leverage properties of the fused activation and quantization functions in the TLUs to further reduce the accumulator size. This is achieved through the _rounded PBS_ feature as described in the [rounded activations and quantizers reference](../explanations/advanced_features.md#rounded-activations-and-quantizers). Adjusting the rounding amount relative to the initial accumulator size can improve latency while maintaining accuracy.
```
Comment (kcelia) [inline docs/deep-learning/optimizing_inference.md:19]: same remark for "you can"

File: docs/deep-learning/fhe_assistant.md [hunk 7]
```diff
@@ -262,14 +154,14 @@ module {
 --------------------------------------------------------------------------------
 ```
 
-There are several calls to `FHELinalg.apply_mapped_lookup_table` and `FHELinalg.apply_lookup_table`. These calls apply PBS to the cells of their input tensors. Their inputs in the listing above are: `tensor<1x2x!FHE.eint<8>>` for the first and last call and `tensor<1x50x!FHE.eint<8>>` for the two calls in the middle. Thus, PBS is applied 104 times.
+In the MLIR code, there are several calls to `FHELinalg.apply_mapped_lookup_table` and `FHELinalg.apply_lookup_table`. These calls apply PBS to the cells of their input tensors. For example, in the code above, the inputs are: `tensor<1x2x!FHE.eint<8>>` for the first and last call and `tensor<1x50x!FHE.eint<8>>` for the two calls in the middle. Thus, PBS is applied 104 times.
```
[the remark this answers is not in the document]
Comment (andrei-stoian-zama) [reply docs/deep-learning/fhe_assistant.md:157]: it's fine I think, there is `apply_lookup_table` and in other MLIRs there might be `apply_mapped_lookup_table`

File: docs/deep-learning/torch_support.md [hunk 8]
```diff
@@ -1,19 +1,26 @@
...
 
-Both approaches require the `rounding_threshold_bits` parameter to be set accordingly. The best values for this parameter need to be determined through experimentation. A good initial value to try is `6`. See [here](../explanations/advanced_features.md#rounded-activations-and-quantizers) for more details.
+  - Use `compile_torch_model` to compile models in this mode.
+
+Both approaches require setting `rounding_threshold_bits` parameter accordingly. You should experiment to find the best values, starting with an initial value of `6`. See [here](../explanations/advanced_features.md#rounded-activations-and-quantizers) for more details.
 
 {% hint style="info" %}
-**See the [common compilation errors page](./fhe_assistant.md#common-compilation-errors) for an explanation of some error messages that the compilation function may raise.**
+See the [common compilation errors page](./fhe_assistant.md#common-compilation-errors) for explanations and solutions of some common errors raised by the compilation function.
```
Comment (kcelia) [inline docs/deep-learning/torch_support.md:20]: I suggest: ... for explanations and solutions to common errors raised by the compilation function

File: docs/deep-learning/optimizing_inference.md (hunk 5 above)
Comment (yuxizama) [reply docs/deep-learning/optimizing_inference.md:11]: We now follow the "Google developer documentation style guide" as a general rule, it's recommended to refer to the reader as "you". See:  https://developers.google.com/style/person

File: docs/deep-learning/torch_support.md [hunk 9]
```diff
@@ -125,18 +132,19 @@ x_test = numpy.array([numpy.random.randn(N_FEAT)])
...
+- `quantized_module.forward(quantized_x, fhe="disable")`: This method computes predictions in the clear on quantized data, and then de-quantize the result. The return value of this function contains the de-quantized (float) output of running the model in the clear. Calling this function on clear data is useful when debugging, but this does not perform actual FHE simulation.
 
 {% hint style="info" %}
-FHE simulation allows to measure the impact of the Table Lookup error on the model accuracy. The Table Lookup error can be adjusted using `p_error`/`global_p_error`, as described in the [approximate computation ](../explanations/advanced_features.md#approximate-computations)section.
+FHE simulation allows to measure the impact of the Table Lookup error on the model accuracy. You can adjust the Table Lookup error using `p_error`/`global_p_error`, as described in the [approximate computation ](../explanations/advanced_features.md#approximate-computations)section.
```
Comment (kcelia) [inline docs/deep-learning/torch_support.md:147]: don't pay attention to this comment. 
I'm just marking the page for me

File: docs/guides/serialization.md [hunk 10]
```diff
@@ -72,26 +74,23 @@ with dumped_model_path.open("w") as f:
...
-serialization. In particular, one cannot serialize a model that was instantiated using callable
-objects for the `train_split` and `predict_nonlinearity` parameters or with `callbacks` being
-enabled.
+Some parameters used for instantiating Quantized Neural Network models are not supported for serialization. In particular, you cannot serialize a model that was instantiated using callable objects for the `train_split` and `predict_nonlinearity` parameters or with `callbacks` being enabled.
 {% endhint %}
 
 ## Loading Models
 
-Loading a built-in model is possible through the following functions:
+You can load a built-in model using the following functions:
```
Comment (kcelia) [inline docs/guides/serialization.md:82]: same with "you can load"

File: docs/guides/client_server.md [hunk 11]
```diff
@@ -69,26 +95,39 @@ encrypted_result = server.run(encrypted_data, serialized_evaluation_keys)
...
+The server-side implementation of a Concrete ML model is illustrated as follows:
 
 ![](../figures/concretemlgraph2.jpg)
 
-The server-side implementation of a Concrete ML model follows the diagram above. The public evaluation keys sent by clients are stored. They are then retrieved for the client that is querying the service and used to evaluate the machine learning model stored in `server.zip`. Finally, the server sends the encrypted result of the computation back to the client.
+The workflow contains the following steps:
+
+1. **Storing the public key**: The public evaluation keys sent by clients are stored.
+1. **Model evaluation**: The public evaluation keys are retrieved for the client that is querying the service and used to evaluate the machine learning model stored in `server.zip`.
+1. S**ending back the result**: The server sends the encrypted result of the computation back to the client.
```
Comment (kcelia) [inline docs/guides/client_server.md:130]: typo: S**ending back the result**

File: docs/guides/hybrid-models.md [hunk 12]
```diff
@@ -1,20 +1,35 @@
...
+- Protects model's Intellectual Property (IP) by keeping the model on a trusted server rather than on client devices.
+
+However, not all applications can be easily converted to FHE computation. The high computation cost of FHE might exceed latency requirements for full conversion.
+
+Hybrid models provide a balance between on-device deployment and cloud-based deployment. This approach involves:
+
+- Executing parts of the model on the client side.
+- Securely processing other parts with FHE on the server side.
+
+Concrete ML supports hybrid deployment for various neural network models, including Multilayer Perceptron (MLP), Convolutional Neural Network (CNN), and Large Language Models.
```
Comment (kcelia) [inline docs/guides/hybrid-models.md:20]: to keep the sentence structure harmonious, you can add: and Large Language Models (LLM).

File: docs/guides/hybrid-models.md [hunk 13]
```diff
@@ -84,7 +103,7 @@ For more information about serving FHE models, see the [client/server section](c
 
 ## Client Side
 
-A client application that deploys a model with hybrid deployment can be developed in a very similar manner to on-premise deployment: the model is loaded normally with PyTorch, but an extra step is required to specify the remote endpoint and the model parts that are to be executed remotely.
+To develop a client application that deploys a model with hybrid deployment, follow a process similar to on-premise deployment. The model is loaded normally with PyTorch, but you need to specify the remote endpoint and the model parts to be executed remotely.
```
Comment (kcelia) [inline docs/guides/hybrid-models.md:106]: I prefer the old version:
A client application that deploys a model with hybrid deployment can be developed in a very similar manner to on-premise deployment: the model is loaded normally with PyTorch, but an extra step is required to specify the remote endpoint and  the model parts to be executed remotely.

File: docs/guides/prediction_with_fhe.md [hunk 14]
```diff
@@ -44,7 +49,9 @@ y_pred_fhe = model.predict(x_test, fhe="execute")
 
 Regarding this LogisticRegression model, as with scikit-learn, it is possible to predict the logits as well as the class probabilities by respectively using the `decision_function` or `predict_proba` methods instead.
 
-Alternatively, it is possible to execute all main steps (key generation, quantization, encryption, FHE execution, decryption) separately.
+### Using separate functions
+
+Alternatively, you can execute key generation, quantization, encryption, FHE execution and decryption separately.
```
Comment (kcelia) [inline docs/guides/prediction_with_fhe.md:54]: same with you

File: docs/guides/client_server.md [hunk 15]
```diff
@@ -69,26 +95,39 @@ encrypted_result = server.run(encrypted_data, serialized_evaluation_keys)
...
+The workflow contains the following steps:
 
-The private data is then encrypted by the client as described in the `serialized_processing.json` file in `client.zip`, and it is then sent to the server. Server-side, the FHE model inference is run on encrypted inputs using the public evaluation keys.
+1. **Key generation**: The client obtains the cryptographic parameters stored in `client.zip` and generates a private encryption/decryption key as well as a set of public evaluation keys.
+1. **Sending public keys**: The public evaluation keys are sent to the server, while the secret key remains on the client.
+1. **Data encryption**: The private data is encrypted by the client as described in the `serialized_processing.json` file in `client.zip`.
+1. **Data transmission**: The encrypted data is sent to the server.
+1. **Encrypted inference**: Server-side, the FHE model inference is run on encrypted inputs using the public evaluation keys.
+1. **Data transmission**:The encrypted result is returned by the server to the client.
```
Comment (kcelia) [inline docs/guides/client_server.md:118]: **Data transmission**: The

File: docs/deep-learning/fhe_assistant.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (kcelia) [reply docs/deep-learning/fhe_assistant.md:157]: But we should update the example, here: 

For example, in the code above, the inputs are: `tensor<1x2x!FHE.eint<8>>` for the first and last call and `tensor<1x50x!FHE.eint<8>>` for the two calls in the middle. Thus, PBS is applied 104 times.

File: docs/deep-learning/optimizing_inference.md (hunk 5 above)
Comment (kcelia) [reply docs/deep-learning/optimizing_inference.md:11]: great thanks, I've deleted the related comments to not spam this RP.

File: docs/deep-learning/fhe_assistant.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (kcelia) [reply docs/deep-learning/fhe_assistant.md:157]: I suggest: For example, in the code above, the inputs are: tensor<1x5x!FHE.eint<15>> for the first and last call. Thus, PBS is applied 2*1*5 times.

CC @andrei-stoian-zama

[the remark this answers is not in the document]
Comment (kcelia) [reply docs/deep-learning/fhe_assistant.md:157]: @yuxizama, here is the updated version:

For example, in the code above, the inputs are: `tensor<1x5x!FHE.eint<15>>` for both the first and last `apply_mapped_lookup_table` call. Thus, the PBS is applied 10 times, corresponding to the size of every encrypted tensor, which is 1x5 multiplied by 2.

File: docs/guides/hybrid-models.md (hunk 13 above)
Comment (yuxizama) [reply docs/guides/hybrid-models.md:106]: In general, we should avoid using past tense in technical documentation. Using active tense and verbs makes sentences easier to read. (See the [style guide](https://developers.google.com/style/voice))

How about
You can develop a client application that deploys a model with hybrid deployment in a very similar manner to on-premise deployment:  Use PyTorch to load the model normally, but specify the remote endpoint and the part of the model to be executed remotely.

Comment (kcelia) [reply docs/guides/hybrid-models.md:106]: I like it

**Review by kcelia — DISMISSED** [review]

**Review by kcelia — APPROVED** [review]
Approved, thanks
